loadimpact/k6-docs
# Breaking changes: mdx support

**Describe the changes**

This PR brings necessary changes to add `mdx` support to this project, making processing of `md` files exactly the same as in `blog` and `main` parts. Along with this:

- `templates/docs` pages logic got completely refactored with new hooks taken from the `blog` and `main`
- fixed all markdown syntax errors that've been breaking the build
- added new component-replacer for adding `ld-json` script to certain doc pages in order to solve build failures that for some reasons `mdx-plugin` was throwing when stumbled upon `ld-json` declaration
- abstracted away components-replacers like `heading-landmark`, `blockquote`, `table-wrapper` etc, improved file structure
- switched `gatsby-plugin-prefetch-google-fonts` (now deprecated) with `gatsby-plugin-google-fonts`


**Notes**

For some reason, Algolia queries on post build can't be finished with the same processing function ported from `blog`, build is just running forever. I've temporarily simplified it to absolute minimum, so be aware of the impact on docs search.  

Unfortunately, there are no way to bring these changes incrementally without seriously affecting the production, so I guess we have to look closely and recheck everything before it hits even the `develop`

**Steps to test**
1. Run the project locally
2. Navigate random pages, use breadcrumbs, click anchor in table of content, make sure everything works exactly the same way

**Screenshots**
N/A

**Issue references**
N/A
